panic: runtime error in version 0.5.5

**Describe the bug**

Livepeer binaries running in `-orchestrator` mode failing with runtime error under specific use-case.

**To Reproduce**

1. Run a B with:
```
./livepeer-linux-amd64/livepeer -network rinkeby -broadcaster -cliAddr 127.0.0.1:7935 -rtmpAddr 127.0.0.1:1935 -httpAddr 0.0.0.0:8935 -orchAddr 127.0.0.1:8936 -transcodingOptions P144p30fps16x9 -depositMultiplier 1 -maxPricePerUnit 5000000000 -pixelsPerUnit 1 -maxTicketEV 1000000000000000000 -v 99
```
2. Wait for the B to report `LPMS Server listening on rtmp://127.0.0.1:1935`

3. Publish content into the B with the command below:
```
ffmpeg -re -f lavfi -i testsrc=size=256x144:rate=30,format=yuv420p -f lavfi -i sine -threads 1 -c:v libx264 -b:v 1000k -preset ultrafast -x264-params keyint=30 -strict -2 -c:a aac -f flv rtmp://127.0.0.1:1935/hello_world
```
4. Run an O with the following command (note, no process is running as `-transcoder`)
```
./livepeer-linux-amd64/livepeer -network rinkeby -orchestrator -cliAddr 127.0.0.1:7936 -httpAddr 127.0.0.1:8936 -serviceAddr 127.0.0.1:8936 -pricePerUnit 2000000000 -pixelsPerUnit 1 -ticketEV 1000000000000000 -orchSecret secret -v 99
```
5. See the following log from the O:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5bd4f9]
```
(Full log output from process below)

**Expected behavior**

In this case, I expect that the O will publish no available sessions to the B, and wait until a T registers to it.

This case models when a O goes offline while a B is already processing content. When the O comes back online, there would be a time when the B is sending to the O, before any Ts have registered.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
```
ubuntu@ip-172-31-43-225:~$ ./livepeer-linux-amd64/livepeer -version
Livepeer Node Version: 0.5.5
Golang runtime version: gc go1.13.4
Architecture: amd64
Operating system: linux
```

**Log**
From `-orchestrator` process which crashed:
```
ubuntu@ip-172-31-43-225:~$ ./livepeer-linux-amd64/livepeer -network rinkeby -orchestrator -cliAddr 127.0.0.1:7936 -httpAddr 127.0.0.1:8936 -serviceAddr 127.0.0.1:8936 -pricePerUnit 2000000000 -pixelsPerUnit 1 -ticketEV 1000000000000000 -orchSecret secret -v 99
I0313 11:11:40.660593    2660 livepeer.go:204] ***Livepeer is running on the rinkeby network: 0xA268AEa9D048F8d3A592dD7f1821297972D4C8Ea***
I0313 11:11:40.661783    2660 db.go:316] Initialized DB node
I0313 11:11:41.177397    2660 accountmanager.go:61] Found existing ETH account
I0313 11:11:41.177425    2660 accountmanager.go:188] Defaulting to first ETH account in keystore 0xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
I0313 11:11:41.177442    2660 accountmanager.go:70] Using Ethereum account: 0xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
I0313 11:11:42.164207    2660 accountmanager.go:93] Please enter the passphrase to unlock Ethereum account 0xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
Passphrase: 
I0313 11:11:46.207982    2660 accountmanager.go:104] Unlocked ETH account: 0xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
I0313 11:11:46.208287    2660 client.go:215] Controller: 0xA268AEa9D048F8d3A592dD7f1821297972D4C8Ea
I0313 11:11:46.324241    2660 client.go:236] LivepeerToken: 0x23b814a57D53b1a7A860194F53401D0D639abED7
I0313 11:11:46.427484    2660 client.go:257] ServiceRegistry: 0xa7AFBB45eb8C27a26379F9C550B3b9be7f09c3eA
I0313 11:11:46.530619    2660 client.go:278] BondingManager: 0xe75a5DccfFe8939F7f16CC7f63EB252bB542FE95
I0313 11:11:46.656612    2660 client.go:299] TicketBroker: 0x40897Ff6e44Ca2e145A691701897844669Bf7cee
I0313 11:11:46.761289    2660 client.go:320] RoundsManager: 0x572d1591bD41f50130FD0212058eAe34F1B17290
I0313 11:11:46.868399    2660 client.go:341] Minter: 0xe5056d1fE48F562dd8AcB1BBFFa147223d448C1B
I0313 11:11:46.977089    2660 client.go:362] LivepeerTokenFaucet: 0x6b752dA302C80795ACc9347AF8f472D3473b19f2
I0313 11:11:47.087770    2660 livepeer.go:469] Price: 2000000000 wei for 1 pixels
 
I0313 11:11:47.831603    2660 livepeer.go:898] Orchestrator 0xXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX is active
I0313 11:11:47.934238    2660 gaspricemonitor.go:134] Cached gas price: 1000000000
I0313 11:11:48.050917    2660 block_watcher.go:318] Backfilling block events (this can take a while)...
I0313 11:11:48.050945    2660 block_watcher.go:319] Start block: 6130376		 End block: 6130375		Blocks elapsed: -1
I0313 11:11:48.057024    2660 block_watcher.go:537] fetching block logs from=6130376 to=6130375
I0313 11:11:48.157951    2660 livepeer.go:777] ***Livepeer Running in Orchestrator Mode***
I0313 11:11:48.158044    2660 mediaserver.go:141] Transcode Job Type: [{P240p30fps16x9 600k 30 426x240 16:9} {P360p30fps16x9 1200k 30 640x360 16:9}]
I0313 11:11:48.158168    2660 cert.go:83] Private key and cert not found. Generating
I0313 11:11:48.158170    2660 webserver.go:70] CLI server listening on 127.0.0.1:7936
I0313 11:11:48.158438    2660 cert.go:22] Generating cert for 127.0.0.1
I0313 11:11:48.158745    2660 rpc.go:155] Listening for RPC on 127.0.0.1:8936
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5bd4f9]

goroutine 105 [running]:
math/big.(*Int).Add(0xc000299a40, 0x0, 0xc00037f360, 0xc0fdf299de)
	/usr/lib/go-1.13/src/math/big/int.go:117 +0x29
github.com/livepeer/go-livepeer/pm.(*recipient).TicketParams(0xc00072a2d0, 0xca870c4c2917c8da, 0xea72b9f45e89a14f, 0xc0fdf299de, 0xc0007a8880, 0x0, 0x0, 0x1)
	/build/pm/recipient.go:221 +0x196
github.com/livepeer/go-livepeer/core.(*orchestrator).TicketParams(0xc0001f0fc0, 0xca870c4c2917c8da, 0xea72b9f45e89a14f, 0xc0fdf299de, 0xc0006a2188, 0x0, 0x0)
	/build/core/orchestrator.go:238 +0xe6
github.com/livepeer/go-livepeer/server.orchestratorInfo(0x1f6dbe0, 0xc0001f0fc0, 0xca870c4c2917c8da, 0xea72b9f45e89a14f, 0xfdf299de, 0xc00068f8c0, 0x16, 0x50, 0x0, 0x0)
	/build/server/rpc.go:256 +0x52
github.com/livepeer/go-livepeer/server.getOrchestrator(0x1f6dbe0, 0xc0001f0fc0, 0xc000128f50, 0x1b6a9e0, 0xc000771800, 0x1f49080)
	/build/server/rpc.go:252 +0x21c
github.com/livepeer/go-livepeer/server.(*lphttp).GetOrchestrator(0xc00028dde0, 0x1f5b200, 0xc000277b30, 0xc000128f50, 0xc00028dde0, 0xc000277b30, 0xc000063a80)
	/build/server/rpc.go:128 +0x41
github.com/livepeer/go-livepeer/net._Orchestrator_GetOrchestrator_Handler(0x1b6a9e0, 0xc00028dde0, 0x1f5b200, 0xc000277b30, 0xc000771860, 0x0, 0x1f5b200, 0xc000277b30, 0xc000780060, 0x59)
	/build/net/lp_rpc.pb.go:1341 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00073eb00, 0x1f67b20, 0xc0001e1320, 0xc0000f7d00, 0xc0001f10e0, 0x2b65080, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:995 +0x460
google.golang.org/grpc.(*Server).handleStream(0xc00073eb00, 0x1f67b20, 0xc0001e1320, 0xc0000f7d00, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1275 +0xd97
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0004934e0, 0xc00073eb00, 0x1f67b20, 0xc0001e1320, 0xc0000f7d00)
	/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xa1
```

--- a/eth/stubclient.go
+++ b/eth/stubclient.go
@@ -165,6 +165,7 @@ type StubClient struct {
 	BlockHashToReturn            common.Hash
 	ProcessHistoricalUnbondError error
 	Orchestrators                []*lpTypes.Transcoder
+	Round                        *big.Int
 	RoundsErr                    error
 	SenderInfo                   *pm.SenderInfo
 	PoolSize                     *big.Int
@@ -182,18 +183,20 @@ type stubTranscoder struct {
 
 func (e *StubClient) Setup(password string, gasLimit uint64, gasPrice *big.Int) error { return nil }
 func (e *StubClient) Account() accounts.Account                                       { return accounts.Account{Address: e.TranscoderAddress} }
-func (e *StubClient) Backend() (Backend, error)                                       { return nil, ErrMissingBackend }
+func (e *StubClient) Backend() (Backend, error)                                       { return nil, nil }
 
 // Rounds
 
-func (e *StubClient) InitializeRound() (*types.Transaction, error)       { return nil, nil }
-func (e *StubClient) CurrentRound() (*big.Int, error)                    { return big.NewInt(0), e.RoundsErr }
-func (e *StubClient) LastInitializedRound() (*big.Int, error)            { return big.NewInt(0), e.RoundsErr }
-func (e *StubClient) BlockHashForRound(round *big.Int) ([32]byte, error) { return [32]byte{}, nil }
-func (e *StubClient) CurrentRoundInitialized() (bool, error)             { return false, nil }
-func (e *StubClient) CurrentRoundLocked() (bool, error)                  { return false, nil }
-func (e *StubClient) CurrentRoundStartBlock() (*big.Int, error)          { return nil, nil }
-func (e *StubClient) Paused() (bool, error)                              { return false, nil }
+func (e *StubClient) InitializeRound() (*types.Transaction, error) { return nil, nil }
+func (e *StubClient) CurrentRound() (*big.Int, error)              { return big.NewInt(0), e.RoundsErr }
+func (e *StubClient) LastInitializedRound() (*big.Int, error)      { return e.Round, e.RoundsErr }
+func (e *StubClient) BlockHashForRound(round *big.Int) ([32]byte, error) {
+	return e.BlockHashToReturn, nil
+}
+func (e *StubClient) CurrentRoundInitialized() (bool, error)    { return false, nil }
+func (e *StubClient) CurrentRoundLocked() (bool, error)         { return false, nil }
+func (e *StubClient) CurrentRoundStartBlock() (*big.Int, error) { return nil, nil }
+func (e *StubClient) Paused() (bool, error)                     { return false, nil }
 
 // Token
 

--- a/eth/watchers/stub.go
+++ b/eth/watchers/stub.go
@@ -266,14 +266,20 @@ func (s *stubSubscription) Err() <-chan error {
 }
 
 type stubBlockWatcher struct {
-	sink chan<- []*blockwatch.Event
-	sub  *stubSubscription
+	sink         chan<- []*blockwatch.Event
+	sub          *stubSubscription
+	latestHeader *blockwatch.MiniHeader
+	err          error
 }
 
 func (bw *stubBlockWatcher) Subscribe(sink chan<- []*blockwatch.Event) event.Subscription {
 	bw.sink = sink
 	bw.sub = &stubSubscription{errCh: make(<-chan error)}
 	return bw.sub
+}
+
+func (bw *stubBlockWatcher) GetLatestBlock() (*blockwatch.MiniHeader, error) {
+	return bw.latestHeader, bw.err
 }
 
 type stubUnbondingLock struct {

--- a/eth/watchers/timewatcher.go
+++ b/eth/watchers/timewatcher.go
@@ -122,6 +122,16 @@ func (tw *TimeWatcher) Watch() error {
 		return fmt.Errorf("error fetching initial transcoderPoolSize err=%v", err)
 	}
 
+	lastSeenBlock, err := tw.watcher.GetLatestBlock()
+	if err != nil {
+		return fmt.Errorf("error fetching last seen block err=%v", err)
+	}
+	blockNum := big.NewInt(0)
+	if lastSeenBlock != nil {
+		blockNum = lastSeenBlock.Number
+	}
+	tw.setLastSeenBlock(blockNum)
+
 	events := make(chan []*blockwatch.Event, 10)
 	sub := tw.watcher.Subscribe(events)
 	defer sub.Unsubscribe()

--- a/eth/watchers/timewatcher_test.go
+++ b/eth/watchers/timewatcher_test.go
@@ -6,6 +6,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/livepeer/go-livepeer/eth"
@@ -39,13 +40,21 @@ func TestSetAndGet_TranscoderPoolSize(t *testing.T) {
 	assert.Equal(size, tw.GetTranscoderPoolSize())
 }
 
-func TestRoundsWatcher_WatchAndStop(t *testing.T) {
+func TestTimeWatcher_WatchAndStop(t *testing.T) {
 	assert := assert.New(t)
 	size := big.NewInt(50)
+	block := big.NewInt(10)
+	round := big.NewInt(1)
+	hash := ethcommon.HexToHash("foo")
 	lpEth := &eth.StubClient{
-		PoolSize: size,
+		PoolSize:          size,
+		BlockNum:          block,
+		BlockHashToReturn: hash,
+		Round:             round,
 	}
-	watcher := &stubBlockWatcher{}
+	watcher := &stubBlockWatcher{
+		latestHeader: &blockwatch.MiniHeader{Number: block},
+	}
 	tw, err := NewTimeWatcher(stubRoundsManagerAddr, watcher, lpEth)
 	assert.Nil(err)
 
@@ -60,6 +69,26 @@ func TestRoundsWatcher_WatchAndStop(t *testing.T) {
 
 	go tw.Watch()
 	time.Sleep(2 * time.Millisecond)
+	// Check state initialization
+	bh := tw.LastInitializedBlockHash()
+	assert.Equal(hash, common.BytesToHash(bh[:]))
+	assert.Equal(round, tw.LastInitializedRound())
+	assert.Equal(size, tw.GetTranscoderPoolSize())
+	assert.Equal(block, tw.LastSeenBlock())
+	tw.Stop()
+
+	// if watcher.GetLatestBlock() == nil, initialise lastSeenBlock to big.NewInt(0)
+	watcher.latestHeader = nil
+	tw, err = NewTimeWatcher(stubRoundsManagerAddr, watcher, lpEth)
+	assert.Nil(err)
+	go tw.Watch()
+	time.Sleep(2 * time.Millisecond)
+	// Check state initialization
+	bh = tw.LastInitializedBlockHash()
+	assert.Equal(hash, common.BytesToHash(bh[:]))
+	assert.Equal(round, tw.LastInitializedRound())
+	assert.Equal(size, tw.GetTranscoderPoolSize())
+	assert.Equal(big.NewInt(0), tw.LastSeenBlock())
 
 	// New Round event
 	watcher.sink <- []*blockwatch.Event{blockEvent}
@@ -71,6 +100,7 @@ func TestRoundsWatcher_WatchAndStop(t *testing.T) {
 	copy(expectedHashForRound[:], newRoundEvent.Data[:])
 	assert.Equal(bhForRound, expectedHashForRound)
 	assert.Equal(size, tw.GetTranscoderPoolSize())
+	assert.Equal(header.Number, tw.LastSeenBlock())
 
 	// Test no NewRound events, values on rw remain the same
 	tw.setTranscoderPoolSize(big.NewInt(10))
@@ -83,17 +113,18 @@ func TestRoundsWatcher_WatchAndStop(t *testing.T) {
 	copy(expectedHashForRound[:], newRoundEvent.Data[:])
 	assert.Equal(bhForRound, expectedHashForRound)
 	assert.Equal(big.NewInt(10), tw.GetTranscoderPoolSize())
+	assert.Equal(header.Number, tw.LastSeenBlock())
 
 	// Test RPC paths (event removed)
 	blockEvent.BlockHeader.Logs = append(blockEvent.BlockHeader.Logs, newRoundEvent)
 	blockEvent.Type = blockwatch.Removed
 	watcher.sink <- []*blockwatch.Event{blockEvent}
 	time.Sleep(2 * time.Millisecond)
-	lastRound = tw.LastInitializedRound()
 	bhForRound = tw.LastInitializedBlockHash()
-	assert.Equal(lastRound.Int64(), int64(0))
-	assert.Equal(bhForRound, [32]byte{})
+	assert.Equal(hash, common.BytesToHash(bh[:]))
+	assert.Equal(round, tw.LastInitializedRound())
 	assert.Equal(size, tw.GetTranscoderPoolSize())
+	assert.Equal(header.Number, tw.LastSeenBlock())
 
 	// Test Stop
 	tw.Stop()
@@ -103,15 +134,15 @@ func TestRoundsWatcher_WatchAndStop(t *testing.T) {
 	// Test watch error when RPC calls fail
 	tw = &TimeWatcher{
 		lpEth: &eth.StubClient{
-			RoundsErr: errors.New("roundswatcher error"),
+			RoundsErr: errors.New("timewatcher error"),
 		},
 	}
 	err = tw.Watch()
 	assert.NotNil(err)
-	assert.Contains(err.Error(), "roundswatcher error")
+	assert.Contains(err.Error(), "timewatcher error")
 }
 
-func TestRoundsWatcher_HandleLog(t *testing.T) {
+func TestTimeWatcher_HandleLog(t *testing.T) {
 	lpEth := &eth.StubClient{}
 	watcher := &stubBlockWatcher{}
 	tw, err := NewTimeWatcher(stubRoundsManagerAddr, watcher, lpEth)
@@ -141,7 +172,9 @@ func TestLastSeenBlock(t *testing.T) {
 
 func TestHandleBlockNum(t *testing.T) {
 	assert := assert.New(t)
-	watcher := &stubBlockWatcher{}
+	watcher := &stubBlockWatcher{
+		latestHeader: &blockwatch.MiniHeader{Number: big.NewInt(1)},
+	}
 
 	tw, err := NewTimeWatcher(stubRoundsManagerAddr, watcher, &eth.StubClient{})
 	assert.Nil(err)
@@ -162,7 +195,9 @@ func TestHandleBlockNum(t *testing.T) {
 
 func TestSubscribeBlocks(t *testing.T) {
 	assert := assert.New(t)
-	watcher := &stubBlockWatcher{}
+	watcher := &stubBlockWatcher{
+		latestHeader: &blockwatch.MiniHeader{Number: big.NewInt(1)},
+	}
 
 	tw, err := NewTimeWatcher(stubRoundsManagerAddr, watcher, &eth.StubClient{})
 	assert.Nil(err)
@@ -188,7 +223,9 @@ func TestSubscribeBlocks(t *testing.T) {
 
 func TestSubscribeRounds(t *testing.T) {
 	assert := assert.New(t)
-	watcher := &stubBlockWatcher{}
+	watcher := &stubBlockWatcher{
+		latestHeader: &blockwatch.MiniHeader{Number: big.NewInt(1)},
+	}
 
 	tw, err := NewTimeWatcher(stubRoundsManagerAddr, watcher, &eth.StubClient{})
 	assert.Nil(err)

--- a/eth/watchers/types.go
+++ b/eth/watchers/types.go
@@ -8,6 +8,7 @@ import (
 
 type BlockWatcher interface {
 	Subscribe(sink chan<- []*blockwatch.Event) event.Subscription
+	GetLatestBlock() (*blockwatch.MiniHeader, error)
 }
 
 type timeWatcher interface {

